### PR TITLE
[Fix] Fix unit test of datasets

### PR DESCRIPTION
**Pull request by JingweiZhang12**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Modification

1. Fix unit test of datasets by regenerating new annotation files.
2. Reduce cuda memory in UT.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
3. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
4. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
5. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by JingweiZhang12** [comment]
I guess `sample_id` in two places: https://github.com/open-mmlab/mmdetection3d/blob/dev-1.x/mmdet3d/evaluation/metrics/kitti_metric.py#L419 and https://github.com/open-mmlab/mmdetection3d/blob/dev-1.x/mmdet3d/evaluation/metrics/kitti_metric.py#L511 don't need to be replaced by `sample_idx` ?

**Comment by JingweiZhang12** [comment]
The package of waymo-open-dataset-tf-2-1-0 does not support the env of python3.9.

File: tests/test_models/test_dense_heads/test_ssn.py [hunk 1]
```diff
@@ -18,6 +18,15 @@ def test_ssn(self):
...
             'ssn/ssn_hv_secfpn_sbn-all_16xb2-2x_nus-3d.py')
+        ssn_cfg.pts_voxel_encoder.feat_channels = [1, 1]
+        ssn_cfg.pts_middle_encoder.in_channels = 1
+        ssn_cfg.pts_backbone.in_channels = 1
+        ssn_cfg.pts_backbone.out_channels = [1, 1, 1]
+        ssn_cfg.pts_neck.in_channels = [1, 1, 1]
+        ssn_cfg.pts_neck.out_channels = [1, 1, 1]
+        ssn_cfg.pts_bbox_head.in_channels = 3
+        ssn_cfg.pts_bbox_head.feat_channels = 1
+        print(ssn_cfg)
```
Comment (Tai-Wang) [inline tests/test_models/test_dense_heads/test_ssn.py:29]: can remove print?

File: requirements/optional.txt [hunk 2]
```diff
@@ -1,2 +1,2 @@
 spconv
-waymo-open-dataset-tf-2-1-0==1.2.0
+# waymo-open-dataset-tf-2-4-0
```
Comment (Tai-Wang) [inline requirements/optional.txt:2]: Why comment this package?

Comment (JingweiZhang12) [reply requirements/optional.txt]: We can't find a waymo-open-dataset-tf that can install on both py3.6 and py3.9 in CI. The waymo-open-dataset-tf > 2-4-0 doesn't support py3.6, and waymo-open-dataset-tf <= 2-4-0 doesn't support py3.9.

**Review by Tai-Wang — APPROVED** [review]

**Pull request merged**